src/TextInput2/TextInput2.tsx

```diff
@@ -92,7 +107,6 @@ const TextInput2: React.FC<Props> = ({
       return;
     }
 
-    const newErrorMessage = errorValidation(value);
```

Comment:
This variable definition is just moved to the new line 96, so that it can be used earlier in case `showErrorOnBlurOnly` is `true`.